src/openzeppelin/account/library.cairo

```diff
@@ -229,29 +229,30 @@ namespace Account:
         let (__fp__, _) = get_fp_and_pc()
         let (tx_info) = get_tx_info()
 
-        # validate transaction        
+        # validate transaction
         with_attr error_message("Account: invalid secp256k1 signature"):
             let (is_valid) = is_valid_eth_signature(tx_info.transaction_hash, tx_info.signature_len, tx_info.signature)
             assert is_valid = TRUE
         end
-                
+
         return _unsafe_execute(call_array_len, call_array, calldata_len, calldata, nonce)
     end
 
     func _unsafe_execute{
             syscall_ptr : felt*,
             pedersen_ptr : HashBuiltin*,
             range_check_ptr,
+            ecdsa_ptr: SignatureBuiltin*,
```

Comment:
And what if we explicitly pass the implicit arguments without including ecdsa_ptr? It looks like it's trying to pass along all the available implicit arguments including those that aren't needed.

Reply:
If I understand you question correctly, I don't think this works. Taking out the implicit `ecdsa_ptr` means we'll still need to create some reference to it to validate the signature...which brings us back to the same initial issue: `Signature hint must point to the signature builtin segment, not {addr}`.

Reply:
My suggestion was to write the call as
```
_unsafe_execute{syscall_ptr, pedersen_ptr, etc...}(call_array_len, call_array, calldata_len, calldata, nonce)
```

Reply:
Afaict the implicit arguments are not only received by `_unsafe_execute` but also passed to the next function call, doing so could revoke `ecdsa_ptr` but other than that it might work.